tests/lib/snaps/test-snapd-udev-input-subsystem/bin/read-evdev-kbd

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -e
+
+# Detect a keyboard by looking in /dev/input/by-path to find the first
+# '-event-kbd'
+evdev_kbd=$(find /dev/input/by-path/ -name '*-event-kbd' -ls | head -1 | sed 's#.* -> ../##')
+
+if [ -z "$evdev_kbd" ]; then
+    echo "No keyboard detected. Aborting"
+    exit 1
+fi
+
+dev="/dev/input/$evdev_kbd"
+if [ ! -c "$dev" ]; then
+    echo "Detected keyboard is not a character device. Aborting"
+    exit 1
+fi
+
+# Obtain an fd for the file
+exec 3<> "$dev"
```

Comment:
https://github.com/snapcore/snapd/compare/master...bboozzoo:bboozzoo/parallel-install

Reply:
EPERM vs EACCES are different depending on if DAC or apparmor is blocking it.